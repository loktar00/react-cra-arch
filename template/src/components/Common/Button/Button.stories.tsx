import * as React from 'react';
import Button from '.';

export const defaultButton = () => {
    return (
        <Button onClick={()=> undefined}>
            Button Text
        </Button>

    );
};

export default {
    title: 'Common/Button'
};
