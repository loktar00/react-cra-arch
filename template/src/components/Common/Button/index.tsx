import React, {forwardRef} from 'react';

const Button = forwardRef(function Button(
    props: React.ComponentPropsWithoutRef<'button'>,
    ref: React.Ref<HTMLButtonElement>
) {
    return (
        <button ref={ref} {...props} className={`da-button ${props.className}`}>
            {props.children}
        </button>
    );
});

Button.displayName = 'DaButton';

export default Button;
