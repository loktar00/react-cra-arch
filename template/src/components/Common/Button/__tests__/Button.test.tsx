import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react';

import Button from '..';

describe('Default Button', () => {
    describe ('Render Button', () => {
        it('It renders the component', () => {
            const tree = renderer.create(
                <Button>Default</Button>).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe ('Button Functionality', () => {
        test('Can click the button', () => {
            const clickHandler = jest.fn();
            const { container }= render(<Button onClick={clickHandler}>Default</Button>);
            const button = screen.getByText('Default');
            fireEvent.click(button);
            expect(clickHandler).toHaveBeenCalledTimes(1);
        });

        test('Cannot click disabled button', () => {
            const clickHandler = jest.fn();
            const { container }= render(<Button disabled onClick={clickHandler}>Default</Button>);
            const button = screen.getByText('Default');
            fireEvent.click(button);
            expect(clickHandler).toHaveBeenCalledTimes(0);
        });
    });
});
