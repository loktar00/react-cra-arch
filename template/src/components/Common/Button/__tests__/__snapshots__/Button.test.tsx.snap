// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Button Render Button It renders the component 1`] = `
<button
  className="da-button undefined"
>
  Default
</button>
`;
